src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -210,10 +229,12 @@ private void CheckInvariants()
                 solutionAttributes,
                 projectIds,
                 options,
+                analyzerReferences,
                 idToProjectStateMap,
                 projectIdToTrackerMap,
                 filePathToDocumentIdsMap,
-                dependencyGraph);
+                dependencyGraph,
+                analyzerReferencesEqual ? _lazyAnalyzers : null);
```

Comment:
This is tricky but we've made this mistake before: be very careful sharing this Lazy<> across snapshots like this, because the lambda can implicitly close over prior Workspace snapshots. (See #41412 for a previous example.) You might not want to use a lambda in the constructor or use some other pattern to avoid the trap.

Reply:
Like so?

```C#
_lazyAnalyzers = lazyAnalyzers ?? CreateLazyHostDiagnosticAnalyzers(analyzerReferences);

// make sure we don't accidentally capture any state but the list of references:
static Lazy<HostDiagnosticAnalyzers> CreateLazyHostDiagnosticAnalyzers(IReadOnlyList<AnalyzerReference> analyzerReferences)
  => new Lazy<HostDiagnosticAnalyzers>(() => new HostDiagnosticAnalyzers(analyzerReferences));
```